### PR TITLE
[dashboard] Use flutter.js and local canvaskit.

### PR DESCRIPTION
**Pull request by ditman**
### Change

This PR updates the dashboard's `web/index.html` to:

* Use `flutter.js` to embed flutter ([as described here](https://docs.flutter.dev/development/platform-integration/web/initialization)).
* Use the `canvasKitBaseUrl` config option to load the locally distributed canvaskit, rather than the version from the external CDN.

### Note

Once [this `flutter/engine` PR](https://github.com/flutter/engine/pull/37187) lands, the dashboard will start to warn (in development) about `flutterConfiguration` being deprecated. The fix is to pass the configuration object to the `initializeEngine` call, like so (feel free to reach out to me for any assistance!):

```javascript
  engineInitializer.initializeEngine({
    canvasKitBaseUrl: "/canvaskit/",
  }).then(function (appRunner) {
    appRunner.runApp();
  });
```

(This also updates the pubspec.lock with whatever was generated last in my machine)

### Fixes

* (Internal security audit)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by CaseyHillers — APPROVED** [review]

**Comment by ditman** [comment]
@CaseyHillers any reason why you are not using the Flutter Service Worker in this app? Is it caching too much? Too little? The index.html was too old? :P

**Review by godofredoc — APPROVED** [review]

**Comment by CaseyHillers** [comment]
> @CaseyHillers any reason why you are not using the Flutter Service Worker in this app? Is it caching too much? Too little? The index.html was too old? :P

This was generated before Flutter web was in beta, feel free to regenerate it to the latest template.

File: dashboard/web/index.html [hunk 1]
```diff
@@ -22,13 +21,25 @@
...
 <body>
-<script src="main.dart.js" type="application/javascript"></script>
 <script>
-  window.flutterConfiguration = {
-    canvasKitBaseUrl: "/canvaskit/"
-  };
+  window.addEventListener('load', function() {
+    // Contact dit@ when this starts throwing warnings in development :)
+    window.flutterConfiguration = {
+      canvasKitBaseUrl: "/canvaskit/"
```
Comment (yjbanov) [inline dashboard/web/index.html:32]: You might want to explain what this does, i.e. uses the local build of CanvasKit rather than the CDN.

**Review by yjbanov — APPROVED** [review]

**Comment by ditman** [comment]
> This was generated before Flutter web was in beta, feel free to regenerate it to the latest template.

Good to know! It's not very needed for this app (makes no sense in offline mode), but it might be worth revisiting later!

File: dashboard/web/index.html (hunk 1 above)
Comment (ditman) [reply dashboard/web/index.html:32]: Yes, that's a useful comment. Added.

**Comment by godofredoc** [comment]
Started a deployment of the flutter-dashboard.

**Pull request merged**